examples/link_pred_ndgck.py

```diff
@@ -0,0 +1,163 @@
+import os.path as osp
+
+import torch
+
+import torch_geometric.transforms as T
+from torch_geometric.datasets import Planetoid
+from torch_geometric.nn import GCNConv
+from torch_geometric.utils import negative_sampling, to_dense_adj
+
+if torch.cuda.is_available():
+    device = torch.device('cuda')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
+
+
+class NdcgK():
```

Comment:
Can we move this to `torch_geometric` directly? For example, in `torch_geometric.metrics`?

Reply:
Yes, of course, I will be doing that.